lib/middleware/jsonValidator.ts

```diff
@@ -1,9 +1,43 @@
-import { Schema, Validator, ValidatorResult } from "jsonschema";
+import { Schema, Validator, ValidatorResult, PreValidatePropertyFunction } from "jsonschema";
 import { NextApiRequest, NextApiResponse } from "next";
 import { MiddlewareRequest, MiddlewareReturn } from "@lib/types";
+import * as htmlparser2 from "htmlparser2";
+import { logMessage } from "@lib/logger";
 
 export type ValidateOptions = {
   jsonKey: string;
+  noHTML?: boolean;
+};
+
+/**
+ * This function when passed into the jsonschema validate function
+ * will receive all possible key value pairs defined in the schema
+ * and will check string values to ensure that there is not HTML
+ * @param object
+ * @param key
+ */
+const htmlChecker: PreValidatePropertyFunction = (object, key) => {
+  const value = object[key];
+  if (key === "subElements") {
+    logMessage.error(object);
+  }
+  if (typeof value === "string") {
+    // we assume all strings can be valid html
+    // use htmlparse2 to construct document
+    // we want to ensure that the only thing in the document is a text element
+    const parsedString = htmlparser2.parseDocument(value);
+
+    // if there are zero children its an empty string, if there is more than one child there is html
+    if (parsedString.children.length === 0) return;
+    else if (parsedString.children.length !== 1) throw new Error(`HTML detected in JSON`);
+
+    const element = parsedString.children[0];
+    // if the parsed element is not of type text its html
+    if (key === "subElements") {
```

Comment:
Is not that some debugging code?

Reply:
Yes good catch I thought I had removed those